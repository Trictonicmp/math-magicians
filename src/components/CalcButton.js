import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
export default class CalcButton extends React.Component {
  render() {
    const { value, clickFn } = this.props;
    return (
      <span
        aria-hidden
        className="calc-button"
        onClick={() => clickFn(value)}
      >
        {value}
      </span>
    );
  }
}

CalcButton.propTypes = {
  value: PropTypes.string,
  clickFn: PropTypes.func.isRequired,
};

CalcButton.defaultProps = {
  value: '',
};