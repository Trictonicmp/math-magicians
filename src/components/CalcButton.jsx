import React from 'react';
import PropTypes from 'prop-types';

const CalcButton = (props) => {
  const { value, clickFn, buttonType } = props;
  return (
    <span
      aria-hidden
      className={`calc-button ${buttonType}`}
      onClick={() => { clickFn(value); }}
    >
      {value}
    </span>
  );
};

CalcButton.propTypes = {
  value: PropTypes.string,
  buttonType: PropTypes.string,
  clickFn: PropTypes.func.isRequired,
};

CalcButton.defaultProps = {
  value: '',
  buttonType: '',
};

export default CalcButton;
