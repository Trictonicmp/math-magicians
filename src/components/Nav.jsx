const Nav = () => (
  <header className="navbar">
    <a href="/" className="icon-link">
      <span>Math</span>
      Magicians
    </a>
    <ul>
      <li><a href="/">Home</a></li>
      <li><a href="/calculator">Calculator</a></li>
      <li><a href="/quotes">Quote</a></li>
    </ul>
  </header>
);

export default Nav;
