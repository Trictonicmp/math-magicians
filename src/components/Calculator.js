import React from 'react';
import CalcButton from './CalcButton';

export default class Calculator extends React.Component {
  constructor() {
    super();
    this.state = {
      inputValue: '0',
    };
  }

  render() {
    const { inputValue } = this.state;
    return (
      <div className="calculator">
        <input type="text" value={inputValue} />
        <div className="buttons-container">
          <CalcButton value="AC" />
          <CalcButton value="+/-" />
          <CalcButton value="%" />
          <CalcButton value="รท" buttonType="action-btn" />
          <CalcButton value="7" />
          <CalcButton value="8" />
          <CalcButton value="9" />
          <CalcButton value="x" buttonType="action-btn" />
          <CalcButton value="4" />
          <CalcButton value="5" />
          <CalcButton value="6" />
          <CalcButton value="-" buttonType="action-btn" />
          <CalcButton value="1" />
          <CalcButton value="2" />
          <CalcButton value="3" />
          <CalcButton value="+" buttonType="action-btn" />
          <CalcButton value="0" buttonType="double-btn" />
          <CalcButton value="." />
          <CalcButton value="=" buttonType="action-btn" />
        </div>
      </div>
    );
  }
}
