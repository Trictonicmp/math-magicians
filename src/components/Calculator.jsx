import React, { useState } from 'react';
import CalcButton from './CalcButton';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculator, setCalculator] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonValue) => {
    setCalculator(calculate(calculator, buttonValue));
  };

  return (
    <div className="calculator">
      <span className="input">
        { calculator.total }
        { calculator.operation }
        { calculator.next }
      </span>
      <div className="buttons-container">
        <CalcButton value="AC" clickFn={handleButtonClick} />
        <CalcButton value="+/-" clickFn={handleButtonClick} />
        <CalcButton value="%" clickFn={handleButtonClick} />
        <CalcButton value="รท" buttonType="action-btn" clickFn={handleButtonClick} />
        <CalcButton value="7" clickFn={handleButtonClick} />
        <CalcButton value="8" clickFn={handleButtonClick} />
        <CalcButton value="9" clickFn={handleButtonClick} />
        <CalcButton value="x" buttonType="action-btn" clickFn={handleButtonClick} />
        <CalcButton value="4" clickFn={handleButtonClick} />
        <CalcButton value="5" clickFn={handleButtonClick} />
        <CalcButton value="6" clickFn={handleButtonClick} />
        <CalcButton value="-" buttonType="action-btn" clickFn={handleButtonClick} />
        <CalcButton value="1" clickFn={handleButtonClick} />
        <CalcButton value="2" clickFn={handleButtonClick} />
        <CalcButton value="3" clickFn={handleButtonClick} />
        <CalcButton value="+" buttonType="action-btn" clickFn={handleButtonClick} />
        <CalcButton value="0" buttonType="double-btn" clickFn={handleButtonClick} />
        <CalcButton value="." clickFn={handleButtonClick} />
        <CalcButton value="=" buttonType="action-btn" clickFn={handleButtonClick} />
      </div>
    </div>
  );
};

export default Calculator;
