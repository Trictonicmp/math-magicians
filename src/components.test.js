import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Nav from './components/Nav';
import CalcButton from './components/CalcButton';
import Calculator from './components/Calculator';

it('nav renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('Calculator renders correctly', () => {
  const tree = renderer
    .create(<CalcButton />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('CalcButton renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
