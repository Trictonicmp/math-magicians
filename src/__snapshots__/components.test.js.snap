// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcButton renders correctly 1`] = `
<div
  className="calculator"
>
  <span
    className="input"
  >
    0
  </span>
  <div
    className="buttons-container"
  >
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      AC
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      +/-
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      %
    </span>
    <span
      aria-hidden={true}
      className="calc-button action-btn"
      onClick={[Function]}
    >
      รท
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      7
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      8
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      9
    </span>
    <span
      aria-hidden={true}
      className="calc-button action-btn"
      onClick={[Function]}
    >
      x
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      4
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      5
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      6
    </span>
    <span
      aria-hidden={true}
      className="calc-button action-btn"
      onClick={[Function]}
    >
      -
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      1
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      2
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      3
    </span>
    <span
      aria-hidden={true}
      className="calc-button action-btn"
      onClick={[Function]}
    >
      +
    </span>
    <span
      aria-hidden={true}
      className="calc-button double-btn"
      onClick={[Function]}
    >
      0
    </span>
    <span
      aria-hidden={true}
      className="calc-button "
      onClick={[Function]}
    >
      .
    </span>
    <span
      aria-hidden={true}
      className="calc-button action-btn"
      onClick={[Function]}
    >
      =
    </span>
  </div>
</div>
`;

exports[`Calculator renders correctly 1`] = `
<span
  aria-hidden={true}
  className="calc-button "
  onClick={[Function]}
/>
`;

exports[`nav renders correctly 1`] = `
<header
  className="navbar"
>
  <a
    className="icon-link"
    href="/"
    onClick={[Function]}
  >
    <span>
      Math
    </span>
    Magicians
  </a>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/maths"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</header>
`;
